pkg/controller/workload/cache/service_cache.go

```diff
@@ -69,3 +92,74 @@ func (s *serviceCache) GetService(resourceName string) *workloadapi.Service {
 	defer s.mutex.RUnlock()
 	return s.servicesByResourceName[resourceName]
 }
+
+// handleWaypoint is used to process a service. If it is a newly added waypoint service, it returns
+// a series of services that need to be updated whose hostname type waypoint address can be converted
+// to IP address type. If it is a service whose waypoint address is waiting to be converted, it is added
+// to the association list of the corresponding waypoint, If it can be converted, it is converted directly,
+// otherwise it waits for the arrival of the waypoint.
+func (s *serviceCache) HandleWaypoint(svc *workloadapi.Service) []*workloadapi.Service {
+	s.mutex.Lock()
+	defer s.mutex.Unlock()
+
+	if len(svc.GetAddresses()) == 0 {
+		return nil
+	}
+	address := svc.GetAddresses()[0]
+	resourceName := svc.ResourceName()
+
+	res := []*workloadapi.Service{}
+	if addr, ok := s.waypointToAddress[resourceName]; ok {
+		// If this svc is a waypoint service, may need updating.
+		log.Infof("Update waypoint %s", resourceName)
+		if addr != nil && addr.String() == address.String() {
+			return nil
+		}
+		s.waypointToAddress[resourceName] = address
+		for _, svc := range s.waypointToServices[resourceName] {
+			s.updateWaypoint(svc, addr)
+			res = append(res, svc)
+		}
+	}
+
+	if svc.GetWaypoint() == nil || svc.GetWaypoint().GetAddress() != nil {
+		return res
```

Comment:
Seems we donot need to return as the svc has been processed originally

Reply:
We don't need to process waypoint with IP address type and `res` is empty in this context.